src/Nethermind/Ethereum2.Bls.Test/YamlNodeExtensions.cs

```diff
@@ -24,29 +24,32 @@ namespace Ethereum2.Bls.Test
 {
     public static class YamlNodeExtensions
     {
-        public static T? Prop<T>(this YamlNode yamlNode, string propertyName) where T : class
+        public static T?[]? ArrayProp<T>(this YamlNode yamlNode, string propertyName) where T : class
         {
             YamlMappingNode? mappingNode = yamlNode as YamlMappingNode;
-            if (mappingNode is null)
-            {
-                return null;
-            }
-            
-            return (T?) Convert.ChangeType((mappingNode[propertyName] as YamlScalarNode)?.Value, typeof(T));
+            var result =
+                (mappingNode?[propertyName] as YamlSequenceNode)?.Children.Select(i => (i as YamlScalarNode)?.Value);
+            return result?.Select(i => i is null ? null : (T) Convert.ChangeType(i, typeof(T))).ToArray();
         }
 
-        public static T?[]? ArrayProp<T>(this YamlNode yamlNode, string propertyName) where T : class
+        public static T?[]? ArrayProp<T>(this YamlNode yamlNode, string propertyName,
+            Func<YamlSequenceNode?, T> converter) where T : class
         {
             YamlMappingNode? mappingNode = yamlNode as YamlMappingNode;
-            var result = (mappingNode?[propertyName] as YamlSequenceNode)?.Children.Select(i => (i as YamlScalarNode)?.Value);
-            return result?.Select(i => i is null ? null : (T) Convert.ChangeType(i, typeof(T))).ToArray();
+            var result = (mappingNode?[propertyName] as YamlSequenceNode)?.Children
```

Comment:
seems to be code style and you can always change it when analysing code locally
I hate vars myself but I am totally fine for other people to use them as they wish. Leave the small pleasures of code style to everyone, otherwise we will leave in a sad world :)